exporter/googlecloudexporter/config.go

```diff
@@ -74,121 +67,42 @@ func (c *Config) setClientOptions() {
 		opts = append(opts, option.WithCredentialsFile(c.CredentialsFile))
 	}
 
-	c.GCPConfig.GetClientOptions = func() []option.ClientOption {
-		return opts
-	}
+	return opts
 }
 
 // createDefaultConfig creates the default config for the exporter
 func createDefaultConfig() config.Exporter {
-	defaultNamespace, _ := os.Hostname()
 	return &Config{
 		ExporterSettings: config.NewExporterSettings(config.NewComponentID(typeStr)),
-		Location:         defaultLocation,
-		Namespace:        defaultNamespace,
 		GCPConfig:        createDefaultGCPConfig(),
 		BatchConfig:      createDefaultBatchConfig(),
-		NormalizeConfig:  createDefaultNormalizerConfig(),
-		DetectorConfig:   createDefaultDetectorConfig(),
-		AttributerConfig: createDefaultAttributerConfig(),
-		TransposerConfig: createDefaultTransposerConfig(),
+		DetectConfig:     createDefaultDetectConfig(),
 	}
 }
 
 // createDefaultGCPConfig creates a default GCP config
-func createDefaultGCPConfig() *gcp.LegacyConfig {
-	gcpFactory := gcp.NewFactory()
-	gcpConfig := gcpFactory.CreateDefaultConfig().(*gcp.LegacyConfig)
-	gcpConfig.RetrySettings.Enabled = false
-	gcpConfig.UserAgent = defaultUserAgent
-	gcpConfig.MetricConfig.Prefix = defaultMetricPrefix
-	gcpConfig.ResourceMappings = []gcp.ResourceMapping{
-		{
-			TargetType: genericNodeResource,
-			LabelMappings: []gcp.LabelMapping{
-				{
-					SourceKey: "host.name",
-					TargetKey: "node_id",
-				},
-				{
-					SourceKey: "location",
-					TargetKey: "location",
-				},
-				{
-					SourceKey: "namespace",
-					TargetKey: "namespace",
-				},
-			},
-		},
-	}
-
-	return gcpConfig
+func createDefaultGCPConfig() *gcp.Config {
+	factory := gcp.NewFactory()
+	config := factory.CreateDefaultConfig().(*gcp.Config)
+	config.RetrySettings.Enabled = false
+	config.UserAgent = defaultUserAgent
+	config.MetricConfig.Prefix = defaultMetricPrefix
+	config.LogConfig.DefaultLogName, _ = os.Hostname()
```

Comment:
Highly unlikely this errors but it'll return an empty string if it does. Is that ok?

Reply:
Yeah. The default value if not set is already an empty string. So this is a noop if that happens. We only attempt to set it here to help customers avoid errors when sending logs. The current exporter will fail to send if the log entry doesn't have an attribute named gcp.log_name _or_ a default log name is not set.